lib/url-state-machine.js

```diff
@@ -533,6 +533,15 @@ function URLStateMachine(input, base, encodingOverride, url, stateOverride) {
   }
   this.input = res;
 
+  // Handle Windows file paths if no state override
+  // Only convert single ASCII letter + :\ pattern (e.g., C:\, D:\)
+  // Note: Only backslash (\), not forward slash (/)
+  // Everything else goes through normal URL parsing
+  if (!stateOverride && !this.url.scheme && /^[a-zA-Z]:\\/u.test(this.input)) {
```

Comment:
This doesn't match the spec text at https://github.com/whatwg/url/pull/874/files#diff-29243b3b9b716b55c6a61970b0c4864f464b139d397fb961a05bb6e1e2b97cabR2251 . Please translate the spec text directly.